uaclient/util.py

```diff
@@ -304,7 +311,8 @@ def which(program: str) -> 'Optional[str]':
     return None
 
 
-def write_file(filename: str, content: str, mode: int = 0o644) -> None:
+def write_file(filename: str, content: 'Union[str, bytes]',
```

Comment:
Instead of complicating the signature of `write_file`, could we introduce a separate bytes-writing function instead (something like `write_file_bytes`)?  (`write_file` could be refactored to be a wrapper around the bytes-writer.)

(This may be moot if we get `gpg` to perform writing for us, of course, because we could also drop these changes.)

Reply:
dropped them. thanks again for that sentiment.